airflow/models/taskinstance.py

```diff
@@ -715,6 +716,13 @@ def _execute_task(task_instance: TaskInstance | TaskInstancePydantic, context: C
             if not task_instance.next_kwargs:
                 task_instance.next_kwargs = {}
             task_instance.next_kwargs[f"{task_to_execute.__class__.__name__}__sentinel"] = _sentinel
+        elif task_instance.next_method == "__trigger_exit__":
+            log.error("Task is resuming from deferral without next_method specified.")
+            raise TaskDeferralError(
+                "Task is resuming from deferral without next_method specified. "
+                "You must either set `method_name` when deferring, or use a trigger "
+                "that is designed to exit the task."
+            )
```

Comment:
```suggestion
            error_msg = "Task is resuming from deferral without next_method specified." 
            log.error(error_msg)
            raise TaskDeferralError(
               f"{error_msg} "
                "You must either set `method_name` when deferring, or use a trigger "
                "that is designed to exit the task."
            )
```

nitpick: not sure whether it's better, but I would probably try this

Reply:
Removed it.